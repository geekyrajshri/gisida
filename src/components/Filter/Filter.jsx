import React from 'react';
import PropTypes from 'prop-types';
import { getDuplicateIndices } from '../../includes/utils';

const Filter = ({ filter,
  view,
  headers,
  sector,
  filters,
  onToggleView,
  checked,
  currentView,
  onFilterSelect = f => f }) =>
  (headers !== undefined && headers.includes(filter)
  && getDuplicateIndices(filters, filter).length === 1 ?
    (<li className={'filter'}>
      <b>{filter}</b>
    </li>) :
    view ?
      (<li className={'filter'}>
        <input
          type="radio"
          id={view}
          name="switch-view"
          value={view}
          checked={view.toLowerCase() === currentView}
          onClick={e => onToggleView(e.target.value.toLowerCase())}
        />
        <label htmlFor={view}>{view}</label>
      </li>) :
      (<li className={'filter'}>
        <input
          type="radio"
          id={sector}
          value={filter}
          name={sector}
          checked={checked.includes(filter)}
          onClick={e => onFilterSelect(sector, filter, e.target.checked, headers)}
        />
        <label htmlFor={filter}>
          {getDuplicateIndices(filters, filter).length > 1 ? <b>{filter}</b> : filter}
        </label>
      </li>));

Filter.propTypes = {
  headers: PropTypes.arrayOf(PropTypes.any),
  filters: PropTypes.arrayOf(PropTypes.any),
  filter: PropTypes.string,
  view: PropTypes.string,
  onToggleView: PropTypes.func,
  sector: PropTypes.string,
  checked: PropTypes.arrayOf(PropTypes.any),
  currentView: PropTypes.string,
  onFilterSelect: PropTypes.func,
};

Filter.defaultProps = {
  headers: [],
  filters: null,
  filter: null,
  view: null,
  onToggleView: null,
  sector: null,
  checked: null,
  currentView: null,
  onFilterSelect: null,
};

export default Filter;
