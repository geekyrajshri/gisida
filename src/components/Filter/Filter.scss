.filter {
  label {
    display: inline-block;
    max-width: 100%;
    margin-bottom: 5px;
    font-weight: normal;
    margin-right: 5px;
    font-size: 13px;

    input[type="radio"] {
      margin-right: 5px;
    }
  }
}
