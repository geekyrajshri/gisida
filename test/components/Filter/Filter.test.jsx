import React from 'react';
import renderer from 'react-test-renderer';
import Filter from '../../../src/components/Filter/Filter';

describe('Filter', () => {
  const onToggleView = jest.fn();
  const onFilterSelect = jest.fn();
  const filter = 'test_filter';
  const view = 'test_view';
  const headers = ['test_headers'];
  const sector = 'test_sector';
  const checked = ['test_checked'];
  const currentView = 'test_currentView';
  it('Filter renders each filter', () => {
    const component = renderer.create(
      <Filter
        onToggleView={onToggleView}
        onFilterSelect={onFilterSelect}
        filter={filter}
        view={view}
        headers={headers}
        sector={sector}
        checked={checked}
        currentView={currentView}
        mapTargetId={'map-1'}
      />);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
