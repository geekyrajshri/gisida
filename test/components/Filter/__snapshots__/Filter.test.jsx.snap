// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Filter renders each filter 1`] = `
<li
  className="filter"
>
  <input
    checked={false}
    id="test_view"
    name="switch-view"
    onClick={[Function]}
    type="radio"
    value="test_view"
  />
  <label
    htmlFor="test_view"
  >
    test_view
  </label>
</li>
`;
